src/zookeeper/zookeeper_session.cpp

```diff
@@ -141,12 +151,26 @@ int zookeeper_session::attach(void *callback_owner, const state_callback &cb)
 {
     utils::auto_write_lock l(_watcher_lock);
     if (nullptr == _handle) {
-        _handle = zookeeper_init(zookeeper_session_mgr::instance().zoo_hosts(),
-                                 global_watcher,
-                                 zookeeper_session_mgr::instance().timeout(),
-                                 nullptr,
-                                 this,
-                                 0);
+        if (dsn::security::FLAGS_enable_zookeeper_kerberos) {
```

Comment:
what will happen when `FLAGS_enable_zookeeper_kerberos=true` but ZK does not support this function?

Reply:
Emm, after this patch, zk client lib has been upgraded to a higher version which support this function already.